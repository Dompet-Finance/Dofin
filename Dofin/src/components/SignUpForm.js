import React from 'react';
import {
  View,
  TextInput,
  Button,
  Text,
} from 'react-native';

class SignUpForm extends React.Component {

  pressTest() {
    return true
  }

  render() {
    return (
      <View>
        <TextInput />
        <TextInput />
        <Button title="" onPress={() => this.pressTest()}/>
      </View>
    )
  }
};

export default SignUpForm;
