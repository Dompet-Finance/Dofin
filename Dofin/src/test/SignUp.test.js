import React from 'react';
import { shallow } from 'enzyme';
import {
  View,
  TextInput,
  Button
} from 'react-native';

import SignUpForm from '../components/SignUpForm';

describe('<SignUp />', () => {
  const wrapper = shallow(<SignUpForm />);

  it('should be a view component', () => {

    expect(wrapper.type()).toEqual(View);
  });

  it('should pass a selected value to the onChange handler', () => {
    const value = 'This is TDD bruh';
    const onChange = jest.fn();
    const wrapper = shallow(
      <TextInput onChange={onChange} />
    );

    expect(wrapper).toMatchSnapshot();

    wrapper.find('TextInput').simulate('change', {
      target: { value },
    });

    expect(onChange).toBeCalledWith({target: {value: 'This is TDD bruh'}});
  });

  it('should render TextInput', () => {

    expect(wrapper.find(TextInput)).toHaveLength(2)
  });

  it('should render Button', () => {

    expect(wrapper.find(Button)).toHaveLength(1)
  });

  test('it should simulate click save on button', () => {
    const button = wrapper.find(Button);
    const prop = button.prop('onPress');
    expect(prop).toBeInstanceOf(Function);
    expect(prop()).toEqual(true)
  })
});
