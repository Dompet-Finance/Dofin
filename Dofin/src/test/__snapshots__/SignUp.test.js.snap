// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUp /> should pass a selected value to the onChange handler 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <TextInput
    onChange={[Function]}
/>,
  "nodes": Array [
    <TextInput
      onChange={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance":  {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TextInput
        onChange={[Function]}
/>,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Component {
        "_getText": [Function],
        "_onBlur": [Function],
        "_onChange": [Function],
        "_onFocus": [Function],
        "_onPress": [Function],
        "_onScroll": [Function],
        "_onSelectionChange": [Function],
        "_onTextInput": [Function],
        "_reactInternalInstance": [Circular],
        "_renderAndroid": [Function],
        "_renderIOS": [Function],
        "_setNativeRef": [Function],
        "blur": [Function],
        "cancelAnimationFrame": [Function],
        "clear": [Function],
        "clearImmediate": [Function],
        "clearInterval": [Function],
        "clearTimeout": [Function],
        "context": Object {
          "focusEmitter": undefined,
          "onFocusRequested": undefined,
        },
        "focus": [Function],
        "isFocused": [Function],
        "measure": [Function],
        "measureInWindow": [Function],
        "measureLayout": [Function],
        "props": Object {
          "onChange": [Function],
        },
        "refs": Object {},
        "requestAnimationFrame": [Function],
        "setImmediate": [Function],
        "setInterval": [Function],
        "setNativeProps": [Function],
        "setTimeout": [Function],
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <TextInput
          onChange={[Function]}
/>,
        "_debugID": 4,
        "_renderedOutput": <TextInput
          onChange={[Function]}
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TextInput
    onChange={[Function]}
/>,
}
`;